packages/server/api/src/app/pieces/piece-metadata-service/hooks/piece-filtering.ts

```diff
@@ -63,3 +73,103 @@ const filterBasedOnCategories = ({
         return categories.some((item) => (p.categories ?? []).includes(item))
     })
 }
+
+
+const searchWithinActionsAndTriggersAsWell = (searchQuery: string, pieces: PieceMetadataSchema[]): PieceMetadataSchema[] => {
+    const putActionsAndTriggersInAnArray = pieces.map((piece) => {
+        const actions = Object.keys(piece.actions).map((name) => piece.actions[name])
+        const triggers = Object.keys(piece.triggers).map((name) => piece.triggers[name])
+
+        return {
+            ...piece,
+            actions,
+            triggers,
+        }
+    })
+
+    const keysInAdditionToActionsAndTriggers = [ 
+        ...defaultFilterKeys,
+        {
+            name: 'actions.displayName',
```

Comment:
If all weights are same why not just?
```ts
    const fuse = new Fuse(putActionsAndTriggersInAnArray, {
        isCaseSensitive: false,
        shouldSort: true,
        keys: ['display', 'description',
            'actiions.displayName', 'actions.description',
            'triggers.description', 'triggers.displayName'],
        threshold: 0.2,
    })
 ```


    
    Let's also remove defaultFilterKeys variable and make it inline as it's more readable